src/utils/transaction.ts

```diff
@@ -671,39 +560,61 @@ export const buildCrossChainAction = async (
               decimals: fromAssetDecimals,
               logoURI: fromAssetIconUrl,
             },
+            toAsset: { decimals: toAssetDecimals },
             amount,
-            offer,
+            receiverAddress,
             accountType,
           },
         } = transactionBlock;
-
+        let transactions: any[] = [];
+        let contractAddress = '0xdf5cFefc1CE077Fc468E3CFF130f955421D9B95a';
         const amountBN = ethers.utils.parseUnits(amount, fromAssetDecimals);
-        const routeData = await fetchBestRoute(
-          sdk,
-          fromChainId,
-          plrDaoAsset.chainId,
-          amountBN,
-          fromAssetAddress,
-          plrDaoAsset.address,
-          sdk.state.accountAddress
-        );
-
-        if (routeData.errorMessage) return { errorMessage: routeData.errorMessage };
-        if (!routeData.bestRoute) return { errorMessage: 'Failed build bridge approval transaction!' };
-
-        let destinationTxns: ICrossChainActionTransaction[] = routeData.destinationTxns || [];
+        
+        if (fromAssetAddress && !addressesEqual(fromAssetAddress, nativeAssetPerChainId[fromChainId].address)) {
+          try {
+            const abi = getContractAbi(ContractNames.ERC20Token);
+            const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, fromAssetAddress);
+            const approvalTransactionRequest = erc20Contract?.encodeApprove?.(contractAddress, amountBN);
+            if (!approvalTransactionRequest || !approvalTransactionRequest.to) {
+              return { errorMessage: 'Failed to build PLR DAO stake approval transaction!' };
+            }
+            const approvalTransaction = {
+              to: approvalTransactionRequest.to,
+              data: approvalTransactionRequest.data,
+              chainId: fromChainId,
+              value: 0,
+              createTimestamp,
+              status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+            };
+            transactions = [...transactions, approvalTransaction];
```

Comment:
in line 569 I see `transactions` being a blank array. Are we manipulating it anywhere prior 589? If not then we may not spread transactions here